sys/riscv/riscv/trap.c

```diff
@@ -538,8 +538,12 @@ do_trap_supervisor(struct trapframe *frame)
 	case SCAUSE_CHERI:
 		if (curthread->td_pcb->pcb_onfault != 0) {
 			frame->tf_a[0] = EPROT;
-			frame->tf_sepc = (uintcap_t)cheri_setaddress(
-			    cheri_getpcc(), curthread->td_pcb->pcb_onfault);
+#if __has_feature(capabilities) && !defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
```suggestion
#ifndef __CHERI_PURE_CAPABILITY__
```
(or !defined(...)) since it's already guarded by `__has_feature(capabilities)`

Reply:
Oh true.  I guess that's the other reason I had done it the other way around initially.